src/test/java/org/sonarsource/scanner/maven/bootstrap/MavenProjectConverterTest.java

```diff
@@ -372,66 +357,72 @@ public void skipOrphanModule() throws Exception {
 
-    assertThat(props.get("sonar.projectKey")).isEqualTo("com.foo:myProject");
-    assertThat(props.get("sonar.projectName")).isEqualTo("My Project");
-    assertThat(props.get("sonar.projectVersion")).isEqualTo("2.1");
-
-    assertThat(props.get("sonar.sources")).isEqualTo(srcMainDir.getAbsolutePath());
+    assertThat(props).containsEntry("sonar.projectKey", "com.foo:myProject")
+      .containsEntry("sonar.projectName", "My Project")
+      .containsEntry("sonar.projectVersion", "2.1")
+      .containsEntry("sonar.sources", srcMainDir.getAbsolutePath());
 
     assertThat(projectConverter.isSourceDirsOverridden()).isTrue();
   }
 
   @Test
-  public void excludeSourcesInTarget() throws Exception {
-    File src2 = temp.newFolder("src2");
-    File pom = temp.newFile("pom.xml");
+  void excludeSourcesInTarget() throws Exception {
+    File src2 = temp.resolve("src2").toFile();
+    src2.mkdir();
+    File pom = temp.resolve("pom.xml").toFile();
+    pom.createNewFile();
 
     MavenProject project = createProject(pom, new Properties(), "jar");
-    project.addCompileSourceRoot(new File(temp.getRoot(), "target").getAbsolutePath());
-    project.addCompileSourceRoot(new File(temp.getRoot(), "src2").getAbsolutePath());
+    project.addCompileSourceRoot(new File(temp.toAbsolutePath().toFile(), "target").getAbsolutePath());
+    project.addCompileSourceRoot(new File(temp.toAbsolutePath().toFile(), "src2").getAbsolutePath());
 
-    Map<String, String> props = projectConverter.configure(Arrays.asList(project), project, new Properties());
+    Map<String, String> props = projectConverter.configure(Collections.singletonList(project), project, new Properties());
 
-    assertThat(props.get("sonar.projectKey")).isEqualTo("com.foo:myProject");
-    assertThat(props.get("sonar.projectName")).isEqualTo("My Project");
-    assertThat(props.get("sonar.projectVersion")).isEqualTo("2.1");
+    assertThat(props).containsEntry("sonar.projectKey", "com.foo:myProject")
+      .containsEntry("sonar.projectName", "My Project")
+      .containsEntry("sonar.projectVersion", "2.1");
 
     assertThat(props.get("sonar.sources").split(",")).containsOnly(src2.getAbsolutePath(), pom.getAbsolutePath());
   }
 
   @Test
-  public void overrideSourcesMultiModuleProject() throws Exception {
+  void overrideSourcesMultiModuleProject() throws Exception {
     Properties pomProps = new Properties();
     pomProps.put("sonar.sources", "src/main");
     pomProps.put("sonar.tests", "src/test");
 
     MavenProject root = createProject(pomProps, "pom");
 
-    File module1BaseDir = temp.newFolder("module1").getAbsoluteFile();
-    File module1SrcDir = temp.newFolder("module1", "src", "main").getAbsoluteFile();
-    File module1TestDir = temp.newFolder("module1", "src", "test").getAbsoluteFile();
+    File module1BaseDir = temp.resolve("module1").toAbsolutePath().toFile();
+    File module1SrcDir = temp.resolve("module1").resolve("src").resolve("main").toAbsolutePath().toFile();
```

Comment:
For 3+ resolutions in a row, you can find them with `(\.resolve\([^\)]++\)){3}`, I think we can simplify here
```suggestion
    File module1SrcDir = temp.resolve(Paths.get("module1", "src", "main")).toAbsolutePath().toFile();
```

Reply:
Could be an idea for a rule